main.cpp

```diff
@@ -395,29 +197,29 @@ int main()
     // FloatType object instanciation and method tests
     FloatType ft (3.0f);
     // --------
-    std::cout << "Initial value of ft: " << *(ft.value) << std::endl;
+    std::cout << "Initial value of ft: " << float(ft) << std::endl;
     // --------
     std::cout << "Use of function concatenation (only float arguments) " << std::endl;
-    std::cout << "New value of ft = (ft + 3.0f) * 1.5f / 5.0f = " << *(ft.add( 3.0f ).multiply(1.5f).divide(5.0f).value) << std::endl;
+    std::cout << "New value of ft = (ft + 3.0f) * 1.5f / 5.0f = " << float( (ft.add( 3.0f ).multiply(1.5f).divide(5.0f)) ) << std::endl;
```

Comment:
FIXME this cast should be unnecessary now that you have User-Defined Conversion functions in place
- [ ] remove C-style cast.